variable/rebin.cpp

```diff
@@ -82,29 +101,49 @@ Variable rebin(const VariableConstView &var, const Dim dim,
     throw except::BinEdgeError(
         "The input does not have coordinates with bin-edges.");
 
-  if (var.dims().inner() == dim) {
-    using namespace rebin_inner_detail;
-    return transform_subspan<std::tuple<
-        args<double, double, double, double>, args<float, float, float, float>,
-        args<float, double, float, double>, args<float, float, float, double>,
-        args<bool, double, bool, double>>>(var.dtype(), dim,
-                                           newCoord.dims()[dim] - 1, newCoord,
-                                           var, oldCoord, core::element::rebin);
+  using transform_args = std::tuple<
+      args<double, double, double, double>, args<float, float, float, float>,
+      args<float, double, float, double>, args<float, float, float, double>,
+      args<bool, double, bool, double>>;
 
+  const bool ascending = is_sorted(oldCoord, dim, SortOrder::Ascending) &&
+                         is_sorted(newCoord, dim, SortOrder::Ascending);
+  if (!ascending && !(is_sorted(oldCoord, dim, SortOrder::Descending) &&
+                      is_sorted(newCoord, dim, SortOrder::Descending)))
+    throw except::BinEdgeError(
+        "Rebin: The old or new bin edges are not sorted.");
```

Comment:
Is this right? - if they are both unsorted you get wrong error message.

Reply:
Do I? Which error do we get then?

Reply:
You get same error. "or" would not be correct. Im not particularly bothered if this gets updated or not as anyone reading the message would surely check both inputs.